### PR TITLE
CE-93: Updating organization validation to collect all errors at once

### PR DESCRIPTION
**Pull request by ialarmedalien**
As title, plus rename the test workflow.

### CODE REVIEW

File: src/us/kbase/workspace/test/database/provenance/OrganizationTest.java [hunk 1]
```diff
@@ -79,32 +79,45 @@ public void buildAndOverwriteOrgIdWithNullOrWhitespace() throws Exception {
 		}
 	}
 
+	private void buildOrganizationFailWithError(final Organization.Builder builder, final String error) {
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/database/provenance/OrganizationTest.java:82]: minor refactor of tests to make a common error checking function

File: src/us/kbase/workspace/test/database/provenance/OrganizationTest.java [hunk 2]
```diff
@@ -48,11 +48,11 @@ public void buildMaximal() throws Exception {
 
 	@Test
 	public void buildTrimOrgNameAndId() throws Exception {
-		for (Map.Entry<String, String> mapElement : VALID_PID_MAP.entrySet()) {
+		for (Map.Entry<String, String> entry : VALID_PID_MAP.entrySet()) {
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/database/provenance/OrganizationTest.java:51]: variable rename for consistency

File: src/us/kbase/workspace/test/database/provenance/OrganizationTest.java [hunk 3]
```diff
@@ -79,32 +79,45 @@ public void buildAndOverwriteOrgIdWithNullOrWhitespace() throws Exception {
...
+			for (String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+				buildOrganizationFailWithError(
+						Organization.getBuilder(nullOrWs).withOrganizationID(invalidPid),
+						"organizationName cannot be null or whitespace only\n" +
+								"Illegal format for organizationID: \"" + invalidPid
+								+ "\"\n" +
+								"It should match the pattern \"^([a-zA-Z0-9][a-zA-Z0-9\\.]+)\\s*:\\s*(\\S.+)$\"");
 			}
 		}
+
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/database/provenance/OrganizationTest.java:121]: will remove this extra line in the next pr

File: src/us/kbase/workspace/database/provenance/Organization.java [hunk 4]
```diff
@@ -63,22 +65,32 @@ public static Builder getBuilder(final String organizationName) {
...
 		}
 
 		/**
-		 * Set the ID of the organization, for example https://ror.org/03rmrcq20.
+		 * Set the ID of the organization, for example ROR:03rmrcq20.
 		 *
-		 * @param organizationID the organization ID. Null or the empty string removes
-		 *                       any current organization ID in the builder.
+		 * @param organizationID
+		 *                the organization ID. Null or the empty string removes
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/database/provenance/Organization.java:84]: formatter fun

**Review by MrCreosote — APPROVED** [review]
one minor comment. 

My view is still that this is going to wind up adding a substantial maintenance burden for a very minor usability improvement, but I don't feel strongly enough about it to spend time arguing

**Pull request merged**